forte/data/data_store.py

```diff
@@ -187,6 +197,76 @@ def _new_tid(self) -> int:
         r"""This function generates a new ``tid`` for an entry."""
         return uuid.uuid4().int
 
+    def _add_new_type(self, type_name: str):
+        """
+        Add a new type input_entry_class into ``_type_attributes``.
+        This method will get all attributes of a type and assign index to them.
+        Args:
+            type_name (str): A string or class type representing a class.
+        Returns:
+            attr_dict (dict): The attribute dict of the given type.
+        Raises:
+            RuntimeError: when the type is not provided by ontology file.
+        """
+        # check if type is in dictionary
+        if type_name in self._type_attributes:
+            return self._type_attributes[type_name]
+        if not self.dynamically_add_type:
+            raise RuntimeError(
+                f"{type_name} is not an existing type in current data store."
+                f"Dynamically add type is disabled."
+            )
+        # get attribute dictionary
+        attributes = self._get_entry_attributes_by_class(type_name)
+
+        attr_dict = {}
+        attr_idx = constants.ENTRY_TYPE_INDEX + 1
+        for attr_name in attributes:
+            attr_dict[attr_name] = attr_idx
+            attr_idx += 1
+
+        self._type_attributes[type_name] = {
+            "attributes": attr_dict,
+            "parent_entry": None,
+        }
+        self.__elements[type_name] = SortedList()
```

Comment:
Do we need to create the list now? This will mean there will be empty list in the `self.__elements` dict, which may make serialization awkward.